karton_humble/karton_humble.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+import json
+import subprocess
+
+from artemis.binds import TaskStatus, TaskType
+from artemis.module_base import ArtemisBase
+from karton.core import Task
+
+
+def process_json_data(result):
+    messages = []
+
+    # Iterate through key-value pairs in the result
+    for key, value in result.items():
+        # Split the key to extract the relevant part
+        key_parts = key.replace("[", "").replace("]", "").split(". ")
+
+        # Check if the key has the expected structure
+        if len(key_parts) >= 2:
+            category = key_parts[1].capitalize()
+
+            # Check if the key is not in the excluded categories and there are relevant values
+            if (key_parts[1].lower() != "info" and
+                    key not in ["[2. Fingerprint HTTP Response Headers]",
+                                "[3. Deprecated HTTP Response Headers/Protocols and Insecure Values]",
+                                "[4. Empty HTTP Response Headers Values]",
+                                "[5. Browser Compatibility for Enabled HTTP Security Headers]"]
+                    and (isinstance(value, dict) or (isinstance(value, list)
+                                                     and any(subvalue for subvalue in value
+                                                             if subvalue and subvalue not in [
+                                                                "Nothing to report, all seems OK!",
+                                                                "No HTTP security headers are enabled."])))):
+                messages.append(f"{category}:")
+
+                # If the value is a dictionary, iterate through subkey-value pairs
+                if isinstance(value, dict):
+                    for subkey, subvalue in value.items():
+                        # Add subkeys and subvalues to messages
+                        if subvalue and subvalue not in ["Nothing to report, all seems OK!",
+                                                         "No HTTP security headers are enabled."]:
+                            messages.append(f"  {subkey}: {subvalue}")
+
+                # If the value is a list, iterate through list items
+                elif isinstance(value, list):
+                    for item in value:
+                        # Add list items to messages
+                        if item and item not in ["Nothing to report, all seems OK!",
+                                                 "No HTTP security headers are enabled."]:
+                            messages.append(f"  {item}")
+
+    return messages
+
+
+class Humble(ArtemisBase):  # type: ignore
+    """
+    Runs humble -> A HTTP Headers Analyzer
+    """
+
+    identity = "humble"
+    filters = [
+        {"type": TaskType.DOMAIN.value},
```

Comment:
if instead of `DOMAIN` tasks you listen on `SERVICE` tasks with `service=HTTP` (as e.g. `bruter` does: https://github.com/CERT-Polska/Artemis/blob/main/artemis/modules/bruter.py#L82) you will catch all HTTP services on all ports, including non-standard ones (if that's your intention)